test_util/marathon.py

```diff
@@ -14,91 +17,140 @@
 log = logging.getLogger(__name__)
 
 
-def get_test_app(custom_port=False):
+class Container(enum.Enum):
+    DOCKER = 'DOCKER'
+    MESOS = 'MESOS'
+    NONE = None
+
+
+class Network(enum.Enum):
+    HOST = 'HOST'
+    USER = 'USER'
+    BRIDGE = 'BRIDGE'
+
+
+class Healthcheck(enum.Enum):
+    HTTP = 'HTTP'
+    MESOS_HTTP = 'MESOS_HTTP'
+
+
+def get_test_app(
+        host_port: int=0,
+        container_port: int=None,
+        container_type: Container=Container.NONE,
+        network: Network=Network.HOST,
+        healthcheck_protocol: Healthcheck=Healthcheck.HTTP,
+        vip: str=None,
+        host_constraint: str=None):
+    """ Creates an app definition for the python test server which will be
+    consistent (i.e. deployable with green health checks and desired network
+    routability). To learn more about the test server, see in this repo:
+    ../packages/dcos-integration-test/extra/util/python_test_server.py
+
+    Args:
+        host_port: port that marathon will use to route traffic into the
+            test server container. If set to zero, then marathon will assign
+            a port (which is referenced by index).
+        container_port: if network is BRIDGE, then the container can have a
+            port remapped inside the container. In HOST or USER network, the
+            container port must be the same as the host port
+        container_type: can be NONE (default Mesos runtime), MESOS (the UCR),
+            or DOCKER
+        health_check_protocol: can be MESOS_HTTP or HTTP
+        vip: either named or unnamed VIP to be applied to the host port
+        host_constraint: string representing a hostname for an agent that this
+            app should run on
+
+    Return:
+        (dict, str): 2-Tuple of app definition (dict) and app ID (string)
+    """
+    if network == Network.BRIDGE:
+        assert container_type == Container.DOCKER, 'BRIDGE network mode only supported for DOCKER container type'
+        if container_port is None:
+            # provide a dummy value for the bridged container port if user is indifferent
+            container_port = 8080
+    else:
+        assert container_port is None or container_port == host_port, 'Cannot declare a different host and '\
```

Comment:
This is intentional. There is no fallback to default. Only when you are using BRIDGE networking does it make any sense to declare a container port. In any other case, there is only one port to decide on an thus container_port MUST equal host_port.

This is an assertion error because if you are running into this, then that means you do not understand what you are trying to do and it should never work.